contrib/opencensus-ext-azure/opencensus/ext/azure/log_exporter/__init__.py

```diff
@@ -110,7 +109,11 @@ def createLock(self):
         self.lock = None
 
     def emit(self, record):
-        self._queue.put(record, block=False)
+        # Convert the raw LogRecord to an envelope before putting it on the
+        # queue as a LogRecord object is not serializable, while an Envelope
+        # object is.
+        envelope = self.log_record_to_envelope(record)
```

Comment:
I think there are performance implications of doing this. We don't want to be mapping every time `logger.X()` is called.

Reply:
There is indeed a performance loss here. However it is not possible to put a raw `LogRecord` on the multiprocessing.Queue as it pickles the object.